src/components/FilterBar/FilterBar.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Box,
+  chakra,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useMultiStyleConfig,
+  Wrap,
+  WrapItem,
+} from "@chakra-ui/react";
+import React, { forwardRef } from "react";
+
+import Button from "../Button/Button";
+import ButtonGroup from "../ButtonGroup/ButtonGroup";
+import Heading from "../Heading/Heading";
+import { LayoutTypes } from "../../helpers/types";
+import MultiSelect from "../MultiSelect/MultiSelect";
+import MultiSelectGroup from "../MultiSelectGroup/MultiSelectGroup";
+import { SelectedItems } from "../MultiSelect/MultiSelect";
+import useNYPLBreapoints from "../../hooks/useNYPLBreakpoints";
+
+interface FilterBarCommonProps {
+  children: React.ReactNode;
+  /** The id of the MultiSelect. */
+  id?: string;
+  /** Determines on Mobile if filter modal overlay is open or closed */
+  isOpen: boolean;
+  /** Heading text of FilterBar */
+  headingText: string;
+  /** Renders the layout of the FilterBar child components in a row or column. */
+  layout?: LayoutTypes;
+  /** The selected items state (items that were checked by user). */
+  selectedItems: SelectedItems;
+  /** User for the global `Clear Filters` button. This button is always shown on mobile, optional on desktop */
+  onClear: () => void;
+  /** If passed, the global `Clear Filters` button will render on desktop and tablet */
+  showClearAll?: boolean;
+}
+
+/** Types related to an optional global submit button `Apply Filters` */
+type SubmitAllProps =
+  | {
+      /** If passed the `Apply Filters` button will render on desktop and tablet */
+      showSubmitAll?: false;
+      /** Function for `Apply Filters` button, needs to be provided by the consuming app */
+      onSubmit?: never;
+    }
+  /** onSubmit is required only if showSubmitAll is passed */
+  | { showSubmitAll: true; onSubmit: () => void };
+
+/** Types related to the Modal handling */
+type HandleModalProps =
+  | {
+      /** If onToggle is passed as prop, it will open and close the filter modal on mobile */
+      onToggle: () => void;
+      /** onOpen and onClose will not be used in this scenario */
+      onOpen?: never;
+      onClose?: never;
+    }
+  | {
+      /** If onToggle is omitted, onOpen and onClose should be provided instead */
+      onToggle?: undefined;
+      /** onOpen will open the filter modal on mobile */
+      onOpen: () => void;
+      /** onClose will close the filter modal on mobile */
+      onClose: () => void;
+    };
+
+export type FilterBarProps = FilterBarCommonProps &
+  SubmitAllProps &
+  HandleModalProps;
+
+/**
+ * `FilterBar` is a wrapper component for filter components.
+ * The DS considers the following components as filter components:
+ * - `MultiSelectGroup`
```

Comment:
Is this currently the only form "group" component that is supported?

Reply:
yes

Reply:
OK, that's what I initially thought.  We'll just need to add support for additional form components before other projects can use the `FilterBar` component.